crates/ruff/src/rules/flake8_comprehensions/fixes.rs

```diff
@@ -462,10 +477,56 @@ pub(crate) fn fix_unnecessary_collection_call(
         }
     };
 
-    Ok(Edit::range_replacement(
-        tree.codegen_stylist(stylist),
-        expr.range(),
-    ))
+    let code = if wrap_in_spaces {
+        wrap_code_in_spaces(tree.codegen_stylist(stylist), checker, expr.range())
+    } else {
+        tree.codegen_stylist(stylist)
+    };
+
+    Ok(Edit::range_replacement(code, expr.range()))
+}
+
+/// Adds spaces around the code generated from `state` if `wrap_in_spaces` is true.
+fn wrap_code_in_spaces(code: String, checker: &Checker, expr_range: TextRange) -> String {
+    if checker.semantic_model().in_f_string() {
+        let str = checker.locator.contents().chars().collect_vec();
+
+        // find the braces that delimit the f-string token
+        let mut si: usize = expr_range.start().into();
+        while str[si] != '{' {
```

Comment:
Would this not detect the wrong braces given that the expression is `a` in: `f"{ {a} }"`?

That would be: an f-string with a single expression, which is a set literal that contains the element `a`.

Reply:
Hmm, yeah it does get this wrong:

```diff
        249 │+23    |-print(f"{ {set(a for a in 'abc')} }")
        250 │+   23 |+print(f"{ { {a for a in 'abc'} } }")
```

But, it might be solvable. Will play around with it.

Reply:
Hmm, it's probably not solvable without a better f-string parser and representation.